### ISSUE TITLE
BUG: overwriting estimators_ 

### ISSUE THREAD

**Issue by Xiaohui9607** [xuyxu/Ensemble-Pytorch#54]
`
        with Parallel(n_jobs=self.n_jobs) as parallel:

        ...
                if test_loader:
                    ...

                        if mse < best_mse:
                            best_mse = mse
                            self.estimators_ = nn.ModuleList()   <---
                            self.estimators_.extend(estimators)
                            if save_model:
                                io.save(self, save_dir, self.logger)

                        msg = ("Epoch: {:03d} | Validation MSE:"
                               " {:.5f} | Historical Best: {:.5f}")
                        self.logger.info(msg.format(epoch, mse, best_mse))

                # Update the scheduler
                with warnings.catch_warnings():
                    warnings.simplefilter("ignore", UserWarning)

                    if self.use_scheduler_:
                        scheduler_.step()

        self.estimators_ = nn.ModuleList()   <---
        self.estimators_.extend(estimators)
        if save_model and not test_loader:
            io.save(self, save_dir, self.logger)
`

**Comment by xuyxu** [comment]
Hi @Xiaohui9607, are you refering to the case that the ensemble aftering calling `fit` actually is not the model with the best validation performance?

**Comment by Xiaohui9607** [comment]
Yes

**Comment by xuyxu** [comment]
Thanks for reporting anyway.
